### PR TITLE
feat(discover): adds a referrer for auth token requests

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
`events` requests authorized by an Auth Token are given an `api.auth-token` snuba referrer

### CODE REVIEW

File: src/sentry/api/endpoints/organization_events.py [hunk 1]
```diff
@@ -42,6 +42,8 @@
     "api.dashboards.worldmapwidget",
 }
 
+API_TOKEN_REFERRER = "api.auth-token"
```
Comment (wmak) [inline src/sentry/api/endpoints/organization_events.py:45]:
```suggestion
API_TOKEN_REFERRER = "api.auth-token.events"
```

File: src/sentry/api/endpoints/organization_events.py [hunk 2]
```diff
@@ -154,7 +156,14 @@ def get(self, request: Request, organization) -> Response:
...
         allow_metric_aggregates = request.GET.get("preventMetricAggregates") != "1"
-        referrer = referrer if referrer in ALLOWED_EVENTS_REFERRERS else "api.organization-events"
+        # Force the referrer to "api.auth-token" for events requests authorized through a bearer token
+        referrer = (
+            API_TOKEN_REFERRER
+            if request.auth
+            else referrer
+            if referrer in ALLOWED_EVENTS_REFERRERS
+            else "api.organization-events"
+        )
```
Comment (wmak) [inline src/sentry/api/endpoints/organization_events.py:166]: nit: bit hard to read once there's multiple ifs like this
```suggestion
        if request.auth:
            referrer = API_TOKEN_REFERRER
        elif referrer not in ALLOWED_EVENTS_REFERRERS:
            referrer = "api.organization-events"
```

**Review by wmak — APPROVED** [review]
LGTM

File: src/sentry/api/endpoints/organization_events.py (hunk 1 above)
Comment (edwardgou-sentry) [reply src/sentry/api/endpoints/organization_events.py]: 👍

**Pull request merged**